### PR TITLE
[ci] removes azure check status from merge gating

### PR DESCRIPTION
**Pull request by iris-garden**
Currently, the `hail-ci-azure` check status is used to determine whether CI should be allowed to merge a PR. This change makes the check not gating for merges, though it will still run.

### CODE REVIEW

**Review by cjllanwarne — APPROVED** [review]
This looks good for now. Longer term I think we should probably change this over to just mirror the github view of "overall check status", ala review states

File: ci/ci/github.py
Comment (cjllanwarne) [inline ci/ci/github.py:56]: Could you possibly document where the status/conclusion enums are defined in the github docs?

Comment (iris-garden) [reply ci/ci/github.py]: added!

File: ci/ci/github.py
Comment (cjllanwarne) [inline ci/ci/github.py:65]: Hmm, SKIPPED for a required check being counted as SUCCESS feels a little wrong to me. Should this be PENDING?

File: ci/ci/github.py
Comment (cjllanwarne) [reply ci/ci/github.py:60]: OK! I think I still prefer the explicit mapping for "less confusing to look at" reasons, but at least we will know if something unexpected goes through now.

Comment (cjllanwarne) [reply ci/ci/github.py:60]: Could we have a unit test with every possible status value from github and assert the correct output is being generated?

**Review by cjllanwarne — CHANGES_REQUESTED** [review]

File: ci/ci/github.py
Comment (iris-garden) [reply ci/ci/github.py]: made it explicit and added a test!

**Review by cjllanwarne — APPROVED** [review]
Fantastic!

**Comment by hail-ci-test-1** [comment]
tested this out in my dev namespace with [this PR](https://github.com/hail-ci-test/ci-test-fhkslcwtu6ij/pull/1) in the associated test repo, which didn't merge until it was approved and had the `ci-test` commit status passing, while the `hail-ci-azure` commit status has failed (the test repo configuration requires one approval and just the `ci-test` commit status to be passing)

**Comment by iris-garden** [comment]
tested this out in my dev namespace with [this PR](https://github.com/hail-ci-test/ci-test-fhkslcwtu6ij/pull/1) in the associated test repo, which didn't merge until it was approved and had the `ci-test` commit status passing, while the `hail-ci-azure` commit status has failed (the test repo configuration requires one approval and just the `ci-test` commit status to be passing)

File: ci/ci/github.py
Comment (chrisvittal) [inline ci/ci/github.py:69]: 1. Use the enum values, like `GithubStatus.SUCCESS`.
2. Return immediately, rather than setting a value and checking it, like so:
```python3
    if state in {"PENDING", "EXPECTED", "ACTION_REQUIRED", "STALE"}:
        return GithubStatus.PENDING
    if state in {"FAILURE", "ERROR", "TIMED_OUT", "CANCELLED", "STARTUP_FAILURE", "SKIPPED"}:
        return GithubStatus.FAILURE
    # etc.
    raise ValueError(f"Unexpected value for GithubStatus: {state}")
```

File: ci/ci/github.py
Comment (chrisvittal) [inline ci/ci/github.py:478]: I find using `-1` as a sentinel confusing. What does this represent? Is there a better value that could be used here?

**Review by chrisvittal — COMMENTED** [review]
Looks good. One suggestion, one request for clarification.

**Review by chrisvittal — CHANGES_REQUESTED** [review]
Wanted to request changes. See previous review.

File: ci/ci/github.py
Comment (iris-garden) [reply ci/ci/github.py]: Very fair! I was initially using `None` to indicate that we haven't received a `review_status` from the GitHub API yet, but it turns out if the repository is not configured to require reviews, the value we receive for it from the API is also `None`. Rewrote it to represent not having received any status from the API as `None`, and having received a status but it was `None` on the API side as `"API_NONE"`.

File: ci/ci/github.py
Comment (iris-garden) [reply ci/ci/github.py]: Done!

File: ci/ci/github.py
Comment (chrisvittal) [inline ci/ci/github.py:65]: You need this for the lints to pass.
```suggestion
    if state in {"FAILURE", "ERROR", "TIMED_OUT", "CANCELLED", "STARTUP_FAILURE", "SKIPPED"}:
        return GithubStatus.FAILURE
    if state in {"SUCCESS", "NEUTRAL"}:
        return GithubStatus.SUCCESS
```

**Review by chrisvittal — APPROVED** [review]
Looks good. There's one change that needs to be made to fix the lints. Also, pytest isn't being found in `test-ci`.

File: ci/ci/github.py
Comment (cjllanwarne) [inline ci/ci/github.py:485]: Did you test the API cost of a round of updates with and without doing this? My naive guess would have been that this is a "fact about a node" rather than "an additional node" so wouldn't cost any more API coins. And it might be simpler not to have this if statement in the middle of the query string?

**Review by cjllanwarne — CHANGES_REQUESTED** [review]
Just one super-minor comment to think about

**Review by cjllanwarne — APPROVED** [review]
Looks great, thanks Iris!

**Pull request merged**
